src/Uno.UI/UI/Xaml/Controls/MenuFlyout/MenuFlyoutPresenter.cs

```diff
@@ -1,13 +1,1008 @@
+		// Can be negative. (-1) means nothing focused.
+		internal int m_iFocusedIndex;
 
-		public MenuFlyoutPresenter() : base()
+		// Weak reference to the menu that ultimately owns this MenuFlyoutPresenter.
+		private WeakReference m_wrOwningMenu;
+
+		// Weak reference to the parent MenuFlyout.
+		private WeakReference m_wrParentMenuFlyout;
+
+		// Weak reference to the owner of this menu.
+		// Only populated if this is the presenter for an ISubMenuOwner.
+		private WeakReference m_wrOwner;
+
+		// Weak reference to the sub-presenter that was created by a child menu owner.
+		private WeakReference m_wrSubPresenter;
+
+		// Whether ItemsSource contains at least one ToggleMenuFlyoutItem.
+		private bool m_containsToggleItems;
+
+		// Whether ItemsSource contains at least one MenuFlyoutItem with an Icon.
+		private bool m_containsIconItems;
+
+		// Whether ItemsSource contains at least one MenuFlyoutItem or ToggleMenuFlyoutItem with KeyboardAcceleratorText.
+		private bool m_containsItemsWithKeyboardAcceleratorText;
+
+		// UNO TODO
+		// private bool m_animationInProgress;
+
+		private bool m_isSubPresenter;
+
+		private int m_depth = 0;
+
+		private FlyoutBase.MajorPlacementMode m_mostRecentPlacement;
+
+		private ScrollViewer m_tpScrollViewer;
+
+		public MenuFlyoutPresenterTemplateSettings TemplateSettings { get; } = new MenuFlyoutPresenterTemplateSettings();
+
+		public MenuFlyoutPresenter()
+		{
+			m_iFocusedIndex = -1;
+			m_containsToggleItems = false;
+			m_containsIconItems = false;
+			m_containsItemsWithKeyboardAcceleratorText = false;
+			// UNO TODO
+			// m_animationInProgress = false;
+			m_isSubPresenter = false;
+			m_mostRecentPlacement = FlyoutBase.MajorPlacementMode.Bottom;
+		}
+
+		internal bool IsSubPresenter { get => m_isSubPresenter; set => m_isSubPresenter = value; }
+
+		// Responds to the KeyDown event.
+
+		protected override void OnKeyDown(KeyRoutedEventArgs pArgs)
+		{
+			var handled = pArgs.Handled;
+
+			if (!handled)
+			{
+				var key = pArgs.Key;
+				pArgs.Handled = KeyPressMenuFlyoutPresenter.KeyDown(key, this);
+			}
+		}
+
+		internal void HandleUpOrDownKey(bool isDownKey)
+		{
+			CycleFocus(isDownKey, FocusState.Keyboard);
+		}
+
+		void CycleFocus(bool shouldCycleDown, FocusState focusState)
+		{
+			IObservableVector<object> spItems;
+			uint nCount = 0;
+
+			// Ensure the initial focus index to validate m_iFocusedIndex when the focused item
+			// is set by application's code like as MenuFlyout Opened event.
+			EnsureInitialFocusIndex();
+
+			var originalFocusedIndex = m_iFocusedIndex;
+
+			var parentFlyout = GetParentMenuFlyout();
+
+			// We should wrap around at the bottom or the top of the presenter if the user isn't using a gamepad or remote.
+			var shouldWrap = parentFlyout != null ? (parentFlyout.InputDeviceTypeUsedToOpen != FocusInputDeviceKind.GameController) : true;
+
+			spItems = Items;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'spItems'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5205492)